src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -98,6 +98,10 @@ private EntityType getFlattenedEntityType(UUID entityTypeId, EntityTypeSource so
     Stream<EntityTypeColumn> childSourceColumns = columns.build().flatMap(Function.identity());
     Stream<EntityTypeColumn> allColumns = Stream.concat(copyColumns(sourceFromParent, originalEntityType, renamedAliases), childSourceColumns);
 
+    if (sourceFromParent != null && Boolean.TRUE.equals(sourceFromParent.getEssentialOnly())) {
+      allColumns = allColumns.filter(EntityTypeColumn::getEssential);
+    }
```

Comment:
Since the `essentialOnly` only comes into play when inheriting columns, I think this should happen inside the above `for`-loop. I think it'd go right after the `stream()` on line 83.
Something like
```java
.filter(col -> !Boolean.TRUE.equals(source.getEssentialOnly()) || col.getEssential())
```

Reply:
Is the column's `essential` property nullable? If so, then you'll need to do the `Boolean.TRUE.equals()` thing for that one, too

Reply:
Yes, it's nullable. Will be adding that  Boolean.TRUE.equals() for it as well